databricks/koalas/indexes.py

```diff
@@ -1494,6 +1494,53 @@ def is_all_dates(self):
         """
         return isinstance(self.spark_type, TimestampType)
 
+    def repeat(self, repeats: int) -> "Index":
+        """
+        Repeat elements of a Index.
+
+        Returns a new Index where each element of the current Index
+        is repeated consecutively a given number of times.
+
+        Parameters
+        ----------
+        repeats : int
+            The number of repetitions for each element. This should be a
+            non-negative integer. Repeating 0 times will return an empty
+            Index.
+
+        Returns
+        -------
+        repeated_index : Index
+            Newly created Index with repeated elements.
+
+        See Also
+        --------
+        Series.repeat : Equivalent function for Series.
+        MultiIndex.repeat : Equivalent function for MultiIndex.
+
+        Examples
+        --------
+        >>> idx = ks.Index(['a', 'b', 'c'])
+        >>> idx
+        Index(['a', 'b', 'c'], dtype='object')
+        >>> idx.repeat(2)
+        Index(['a', 'b', 'c', 'a', 'b', 'c'], dtype='object')
+        """
+        if not isinstance(repeats, int):
+            raise ValueError("`repeats` argument must be integer, but got {}".format(type(repeats)))
+        elif repeats < 0:
+            raise ValueError("negative dimensions are not allowed")
+
+        sdf = self._internal.sdf.select(self._internal.scol)
+        internal = _InternalFrame(
+            sdf=sdf, index_map=[(sdf.columns[0], self._internal.index_names[0])]
```

Comment:
I'm just curious about this, why just don't use `self._internal.index_map` here rather create index_map again manually? 🤔 (i guess maybe do they works in the same way ?)

Reply:
because `self._internal.scol` could have a different name.

Reply:
@ueshin Ah.. I think found the case. Thanks!! :)

```python
>>> self = ks.Series(['a', 'b', 'c'], index=[1, 2, 3])
>>> sdf = self._internal.sdf.select(self._internal.scol)
>>> self._internal.index_map
[('__index_level_0__', None)]
>>> [(sdf.columns[0], self._internal.index_names[0])]
[('0', None)]
```

Reply:
Actually that's a `Series` case.
The example for `Index` is like:

```py
>>> import databricks.koalas as ks
>>> kidx = ks.Index([1,2,3])
>>> kidx1 = (kidx + 1)
>>> kidx1._internal.index_scols, kidx1._internal.scol
([Column<b'__index_level_0__'>], Column<b'(__index_level_0__ + 1)'>)
>>> kidx1._internal.sdf.select(kidx1._internal.index_scols + [kidx1._internal.scol])
DataFrame[__index_level_0__: bigint, (__index_level_0__ + 1): bigint]
```

Reply:
But actually you raised a good point.
Maybe we should revisit the infrastructure of `Index`.
Let me think.

Reply:
Oops... sorry I forgot extracting `Index` from `Series` when i tested.

Thanks for the example and I totally agree that we consolidate them! 😺

Reply:
Ah, the example you replied is super good. Thanks again!!